blocks/model.py

```diff
@@ -161,15 +162,18 @@ def __init__(self, outputs):
             raise ValueError("top bricks with the same name:"
                              " {}".format(', '.join(repeated_names)))
 
-        brick_param_names = {
-            v: k for k, v in Selector(self.top_bricks).get_params().items()}
-        self.params = []
-        for param in VariableFilter(roles=[PARAMETER])(self.shared_variables):
-            if param in brick_param_names:
-                self.params.append((brick_param_names[param], param))
+        brick_parameter_names = {
+            v: k
+            for k, v in Selector(self.top_bricks).get_parameters().items()}
+        self.parameters = []
+        for parameter in VariableFilter(
+                roles=[PARAMETER])(self.shared_variables):
+            if parameter in brick_parameter_names:
+                self.parameters.append((brick_parameter_names[parameter],
+                                        parameter))
             else:
-                self.params.append((param.name, param))
-        self.params = OrderedDict(self.params)
+                self.parameters.append((parameter.name, parameter))
+        self.parameters = OrderedDict(self.parameters)
```

Comment:
You should not try to set self.parameters anymore. It's ComputationGraph's property. There is no need to build it here anymore.